File: src/evai/command/BaseCommands.java
package evai.command;

import mindustry.*;
import mindustry.gen.*;
import mindustry.world.*;
import mindustry.world.blocks.environment.*;
import java.util.*;

public class BaseCommands implements CommandRegistry.CommandHandler {
    @Override
    public CommandResult handle(String action, String[] args) {
        switch (action) {
            case "help": return help(args);
            case "status": return status();
            case "whereami": return whereami();
            case "clear": return new CommandResult(true, "chat cleared");
            case "version": return new CommandResult(true, "Eve Assistant Mod v1.0");
            default: return new CommandResult(false, "unknown base command: " + action);
        }
    }

    private CommandResult help(String[] args) {
        return new CommandResult(true, "Eve Assistant Mod v1.0\n" +
            "Commands: help, status, whereami, clear, version\n" +
            "list <type> [filter], info <name>\n" +
            "scan [r], scan resources/liquids, check <field> <x> <y>, map\n" +
            "debug <setting> [value]\n" +
            "set <x> <y> <field> <value>, nuke <x> <y> [r]\n" +
            "build <block> <x> <y> [r], build area/line/ring\n" +
            "break <x> <y>, break area <x1> <y1> <x2> <y2>\n" +
            "schem <create/place/list/delete/rename/export/import>\n" +
            "spawn <unit> <x> <y> [team] [amt]\n" +
            "control [unit], kill <all/enemies/allies> [r]\n" +
            "listen/fire <event>, js/logic/eval <code>\n" +
            "save/load <name>, save list/delete/current\n" +
            "export blocks/state/tile/units");
    }

    private CommandResult status() {
        Map<String, Object> data = new HashMap<>();
        data.put("map", Vars.state.map != null ? Vars.state.map.name() : "none");
        data.put("wave", (int) Vars.state.wave);
        data.put("team", Vars.player != null ? Vars.player.team().name : "none");
        data.put("player", Vars.player != null ? Vars.player.name : "none");
        data.put("tiles", Vars.world.width() + "x" + Vars.world.height());
        data.put("mode", Vars.state.rules != null ? Vars.state.rules.mode().name() : "unknown");
        return new CommandResult(true, "", data);
    }

    private CommandResult whereami() {
        if (Vars.player == null) return new CommandResult(false, "no player");
        Player p = Vars.player;
        Tile tile = Vars.world.tileWorld(p.x, p.y);
        Map<String, Object> data = new HashMap<>();
        data.put("x", (int) p.x);
        data.put("y", (int) p.y);
        data.put("tileX", tile != null ? tile.x : -1);
        data.put("tileY", tile != null ? tile.y : -1);
        data.put("block", tile != null && tile.block() != null ? tile.block().name : "none");
        data.put("floor", tile != null && tile.floor() != null ? tile.floor().name : "none");
        return new CommandResult(true, "", data);
    }
}

File: src/evai/command/BaseCommands.kt
package evai.command

import mindustry.*
import evai.command.CommandRegistry.CommandHandler

class BaseCommands : CommandHandler {
    override fun handle(action: String, args: Array<String>): CommandResult = when (action) {
        "help" -> {
            if (args.isNotEmpty()) {
                CommandResult(true, "See /eve <action> [args]. Full docs in PLAN.md")
            } else {
                CommandResult(true, """Eve Assistant Mod v1.0
Commands:
  help, status, whereami, clear, version
  list <type> [filter], info <name>
  scan [r], scan resources/liquids, check <field> <x> <y>, map
  debug <setting> [value]
  set <x> <y> <field> <value>, set items/liquid/power/health/team/config x y <val>
  nuke <x> <y> [r]
  build <block> <x> <y> [r], build area/line/ring <block> <args...>
  break <x> <y>, break area <x1> <y1> <x2> <y2>
  schem <create/place/list/delete/rename/export/import> <args...>
  spawn <unit> <x> <y> [team] [amt], spawn enemy/ally <unit> <x> <y> [amt]
  control [unit], kill <all/enemies/allies> [r]
  listen <event>, listen all/stop/list, fire <event> [args]
  js <code>, logic <code>, eval <expr>
  save/load <name>, save list/delete/current
  export blocks/state/tile/units [filter/r]""")
            }
        }
        "status" -> {
            val state = Vars.state
            val world = Vars.world
            val player = Vars.player
            CommandResult(true, data = mapOf(
                "map" to (state.map?.name() ?: "none"),
                "wave" to state.wave,
                "team" to (player?.team()?.name ?: "none"),
                "player" to (player?.name ?: "none"),
                "tiles" to "${world.width()}x${world.height()}",
                "mode" to (state.rules?.mode()?.name ?: "unknown")
            ))
        }
        "whereami" -> {
            val player = Vars.player
            if (player != null) {
                val x = player.x.toInt()
                val y = player.y.toInt()
                val tile = Vars.world.tileWorld(x.toFloat(), y.toFloat())
                CommandResult(true, data = mapOf(
                    "x" to x, "y" to y,
                    "tileX" to (tile?.x ?: -1),
                    "tileY" to (tile?.y ?: -1),
                    "block" to (tile?.block()?.name ?: "none"),
                    "floor" to (tile?.floor()?.name ?: "none")
                ))
            } else {
                CommandResult(false, "no player")
            }
        }
        "clear" -> CommandResult(true, "chat cleared")
        "version" -> CommandResult(true, "Eve Assistant Mod v1.0")
        else -> CommandResult(false, "unknown base command: $action")
    }
}